monkey/common/agent_events/propagation_event.py

```diff
@@ -0,0 +1,21 @@
+from ipaddress import IPv4Address
+
+from pydantic import Field
+
+from . import AbstractAgentEvent
+
+
+class PropagationEvent(AbstractAgentEvent):
+    """
+    An event that occurs when the Agent propagates on a host
+
+    Attributes:
+        :param success: Status of the propagation
+        :param exploiter_name: Name of the exploiter that propagated
+        :param error_message: Message if an error occurs during propagation
+    """
+
+    target: IPv4Address
```

Comment:
Do we want to document this field as well?

Reply:
As we are overriding the field, I don't think it is really needed.